server/base/src/main/java/org/apache/accumulo/server/security/handler/ZKSecurityTool.java

```diff
@@ -114,16 +119,34 @@ public static byte[] createPass(byte[] password) throws AccumuloException {
     return cryptHash.getBytes(UTF_8);
   }
 
+  private static final Cache<String,String> CRYPT_PASSWORD_CACHE = Caffeine.newBuilder()
+      .scheduler(Scheduler.systemScheduler()).expireAfterWrite(3, TimeUnit.SECONDS).build();
```

Comment:
> For this to be of benefit, we'd have to be creating a lot of connections pretty quickly. I'm not sure how realistic that would be.

I think that is what is described in #2700 as part of a test. I could see real work implications, typically on application startup where you have a thundering herd type situation.

> Wouldn't we want expireAfterAccess rather than write? If it's still frequently being accessed, it'd be good to keep it around, regardless of the last time it was written to the cache.

This was based on an earlier comment about have really short times in the cache. I think at this point it could be longer and based on access rather than write.

> I'm not sure this is worth it. The underlying issue seems highly dependent on specific JVM versions, and specific JCE providers and their native instruction hardware optimizations.

I would agree, but I would also say that we are doing work that we might not have to do.

Reply:
In 67e6332 I changed the expiration criteria from afterWrite to afterAccess, put a max size on the cache to 64, and changed the expiration time from 3s to 1m based on a comment earlier by @keith-turner

Reply:
> > For this to be of benefit, we'd have to be creating a lot of connections pretty quickly. I'm not sure how realistic that would be.
> 
> I think that is what is described in #2700 as part of a test. I could see real work implications, typically on application startup where you have a thundering herd type situation.

I suppose. I'm just not sure how well the observations of lock contention or brief periods of high CPU usage translate to end user experience, or overall user application performance. This isn't much complexity, but in general, I'm resistant to specialized code in our application that tries to work around performance limitations in user hardware/external libraries, just because of separation of responsibilities. I'm also resistant to workarounds that cater to specific deployment use cases when better deployment decisions (like choosing a better JCE library or deploying using hardware with crypto extensions) could also address the problem. This hits on both those points, and it's not clear it's even noticeable to users. But, the workaround is simple enough, so maybe it's okay.